util/exercise/src/cmd/templates/test_fn.rs

```diff
@@ -1,6 +1,9 @@
-{% import "macros.rs" as macros %}
+{% import "macros.rs" as macros -%}
+
 #[test]
+{% if not dont_ignore -%}
```

Comment:
I'm not familiar enough with Tera to know if this is possible, but could you set the global `dont_ignore` here, instead of up in `test_file.rs`? Like this:

```
{% if first_test_case -%}
{% set_global first_test_case = false -%}
{% else -%}
#[ignore]
{% endif -%}
```

Reply:
Part of the goal of this change suggestion would be to simplify case generation: for each case, there would only be a call to `{{ macros::gen_test_fn(case=item, dont_ignore=dont_ignore) }}`, because `dont_ignore` would be updated here, and the properties will be computed elsewhere.

Reply:
That's a really good idea. Unfortunately, Keats/tera#336 mentions that macros can't interact with the global context, and I can confirm it after trying it out.

That being said, I really like your idea. Since the macro is the stumbling block, I'm thinking we could remove it and set up the variables "manually". This implies a bit of duplication (iterating over a group of cases vs a single case), but it's only a few lines and the duplicates remain next to each other in a relatively small file.

I haven't changed the name of the global variable yet, since the two changes would fit together nicely in a single commit.